packages/web3-eth-contract/src/encoding.ts

```diff
@@ -233,3 +235,27 @@ export const decodeMethodReturn = (abi: AbiFunctionFragment, returnValues?: HexS
 
 	return result;
 };
+
+export const decodeErrorData = (errorsAbi: AbiErrorFragment[], error: Eip838ExecutionError) => {
+	if (error?.data) {
+		let errorName: string | undefined;
+		let errorSignature: string | undefined;
+		let errorArgs: { [K in string]: unknown } | undefined;
+		try {
+			const errorSha = error.data.slice(0, 10);
+			const errorAbi = errorsAbi.find(abi => encodeErrorSignature(abi).startsWith(errorSha));
+
+			if (errorAbi?.inputs) {
+				errorName = errorAbi.name;
+				errorSignature = jsonInterfaceMethodToString(errorAbi);
+				// decode abi.inputs according to EIP-838
+				errorArgs = decodeParameters([...errorAbi.inputs], error.data.substring(10));
+			}
+		} catch (err) {
+			console.error(err);
```

Comment:
should throw error if decoding fails for any reason

Reply:
Actually, the ABI would not be available all cases. And even it could appear to be conflicting with the raised error. The reason for this is that the smart contract function could call another function(s) that exists in a different smart contract(s), which could throw the error. In this case, the exact ABI for that subsequent call(s) is not available. Furthermore, in such a case, the error that could raise could even have for example the same error name but with different parameters or parameters' types.
For this, we need to be tolerant and just pass the error data even if an exception has been raised while we are trying to do so.